glib/src/object.rs

```diff
@@ -1073,6 +1073,14 @@ glib_object_wrapper!(@object
 );
 pub type ObjectClass = Class<Object>;
 
+pub struct ObjectGuard<'a, T: ObjectType>(&'a T);
```

Comment:
```suggestion
#[must_use = "if unused the property notifications will immediately be thawed"]
pub struct PropertyNotificationFreezeGuard<'a, T: ObjectType>(&'a T);
```

Also not sure if this should take a reference or clone the object. There are probably cases where you want to e.g. send such a thing through a channel or keep it during async operations until they're completely done. A reference would make that all a bit tricky.

Reply:
Also you can make it non-generic by storing a `glib::Object` internally. That keeps things a bit simpler too.